lib/engine/step/g_18_al/buy_sell_par_shares.rb

```diff
@@ -6,11 +6,19 @@ module Engine
   module Step
     module G18AL
       class BuySellParShares < BuySellParShares
+        PURCHASE_ACTIONS = [Action::BuyCompany, Action::BuyShares, Action::Par].freeze
+
         def setup
           super
           @controller_at_start = current_controller
         end
 
+        def actions(entity)
+          actions = super
+          actions << 'buy_company' if actions.include?('buy_shares')
```

Comment:
hm, shouldn't you only add buy_company action if there are companies to buy? then you wouldn't need to do all those checks in the view that you do

Reply:
Rewritten this quite a bit, but now the buy_company will only be available if there is a legal buy.